### PR TITLE
Created basic models

### PR DESCRIPTION
**Pull request by chriswright47**
Made basic models. Wired up associations and wrote test for user model (more tests are on the way)
@mikelikesbikes @ryanbriones

### CODE REVIEW

File: .rspec [hunk 1]
```diff
@@ -0,0 +1 @@
+--color
```
Comment (ryanbriones) [inline .rspec:1]: I wouldn't commit this file unless you want every person to explicitly use the same RSpec settings

File: app/models/post.rb [hunk 2]
```diff
@@ -0,0 +1,14 @@
+class Post < ActiveRecord::Base
+  attr_accessible :body, :question_id, :title, :user_id
+
+  has_many :votes
+  has_many :comments
+  has_many :taggings
+  has_many :tags, :through => :taggings
+
+  has_many :answers, class_name: "Post", foreign_key: "question_id"
```
Comment (ryanbriones) [inline app/models/post.rb]: Does this work? I don't understand what is going on here.

**Comment by ryanbriones** [comment]
Can you show some example code of how your post model would be used? I've having a hard time understanding the relationships.

**Comment by ryanbriones** [comment]
Personally I'm not a fan of creating all your models all at once. You should be adding models as the need emerges.

I think that could be a source of the complexity in the Post model.

**Comment by chriswright47** [comment]
Changed models and migrations to accommodate polymorphic associations, instead of self join.  Questions and answers are now voteable and commentable. @ryanbriones @mikelikesbikes

**Comment by ryanbriones** [comment]
These associations should be covered by tests by tomorrow.

https://github.com/thoughtbot/shoulda-matchers FYI

**Pull request merged**